bypros/proxy.go

```diff
@@ -0,0 +1,299 @@
+package bypros
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/gorilla/websocket"
+	// register the sql driver
+	_ "github.com/jackc/pgx/stdlib"
+	"go.dedis.ch/cothority/v3/bypros/browse/paginate"
+	"go.dedis.ch/cothority/v3/byzcoin"
+	"go.dedis.ch/cothority/v3/skipchain"
+	"go.dedis.ch/onet/v3/log"
+	"go.dedis.ch/protobuf"
+	"golang.org/x/xerrors"
+)
+
+const (
+	pingWsInterval = time.Minute * 1
+
+	defaultPageSize = 200
+	defaultNumPages = 40
+)
+
+// Follow starts following a node, which means it will listen to every new
+// blocks and update the database accordingly.
+func (s *Service) Follow(req *Follow) (*EmptyReply, error) {
+	select {
+	case <-s.follow:
+		log.LLvl1("proxy following")
+	default:
+		return nil, xerrors.Errorf("already following")
+	}
+
+	s.following = true
+	s.stopFollow = make(chan struct{}, 1)
+
+	waitDone := sync.WaitGroup{}
+
+	apiEndpoint, err := getWsAddr(req.Target)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get ws addr: %v", err)
+	}
+
+	apiURL := fmt.Sprintf("%s/%s/%s", apiEndpoint, byzcoin.ServiceName, "StreamingRequest")
+	c, _, err := websocket.DefaultDialer.Dial(apiURL, nil)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to dial %s: %v", apiURL, err)
+	}
+
+	streamReq := byzcoin.StreamingRequest{
+		ID: req.ScID,
+	}
+
+	buf, err := protobuf.Encode(&streamReq)
```

Comment:
Why don't you use `byzcoin.Client.StreamTransactions` here?

Reply:
1) To have a connection that stays alive beyond the hard-coded 5 minutes of the cothority client  
2) To remove a source of uncertainty

Reply:
1. If you add a new parameter to `onet.Options`, you might also change this
  - Having connections time out from time to time is not so bad - as long as you reconnect
  - e.g., the dynacred-library happily reconnects the streaming listener - might be even every 30 seconds...
2. Why uncertiainty?

Reply:
2: onet has its own way of handling ws connections, which was supposedly raising some errors and making the tests (more) unstable. I came to a point where I didn't have time to fight it anymore.